content/en/blog/_posts/2022-12-06-non-graceful-node-shutdown-to-beta.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: blog
+title: "Kubernetes 1.26: Non-Graceful Node Shutdown Moves to Beta"
+date: 2022-12-06T10:00:00-08:00
+slug: kubernetes-1-26-non-graceful-node-shutdown-beta
+---
+
+**Author:** Xing Yang and Ashutosh Kumar (VMware)
+
+Kubernetes v1.24 introduces [alpha support](https://kubernetes.io/blog/2022/05/20/kubernetes-1-24-non-graceful-node-shutdown-alpha/) for [Non-Graceful Node Shutdown](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2268-non-graceful-shutdown).
+In Kubernetes v1.26, this feature moves to beta. This feature allows stateful workloads to failover to a different node after the original node is shutdown or in a non-recoverable state such as the hardware failure or broken OS.
+
+## What is a Node Shutdown in Kubernetes
+
+In a Kubernetes cluster, it is possible for a node to shutdown. This could happen either in a planned way or it could happen unexpectedly. You may plan for a security patch, a kernel upgrade, and need to reboot the node; or a node may shutdown due to preemption of VM instances; or a node may shutdown due to a hardware failure, or a software problem, etc.
+
+To trigger a node shutdown, you could run a shutdown or poweroff command, or physically press a button to power off a machine.
+
+A node shutdown could lead to workload failure if the node is not drained before the shutdown. A node shutdown can be either graceful or non-graceful.
+
+## What is a Graceful Node Shutdown
+
+The [Graceful Node Shutdown](https://kubernetes.io/blog/2021/04/21/graceful-node-shutdown-beta/) feature allows Kubelet to detect a node shutdown event, properly terminate the pods, and release resources, before the actual shutdown. Critical pods will be terminated after all the regular pods are terminated, to make sure the critical function of an application can work as long as possible.
+
+## What is a Non-Graceful Node Shutdown
+
+A Node Shutdown can be `Graceful` only if the node shutdown action can be detected by Kubelet’s Node Shutdown Manager. However, there are cases where a node shutdown action may not be detected by Kubelet's Node Shutdown Manager. This could happen either because the shutdown command does not trigger the inhibitor locks mechanism used by Kubelet or because of a user error, i.e., the `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` are not configured properly.
```

Comment:
Why are we capitalizing Kubelet?

```suggestion
A Node Shutdown can be `Graceful` only if kubelet’s Node Shutdown Manager can detect the node shutdown action. However, there are cases where kubelet's Node Shutdown Manager may not detect a node shutdown action. This could happen because the shutdown command does not trigger the [Inhibitor Locks](https://www.freedesktop.org/wiki/Software/systemd/inhibit) mechanism used by kubelet or because of a user error, i.e., the `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` are not configured correctly.
```

Reply:
Done